manifests/init.pp

```diff
@@ -51,25 +55,28 @@
 # @author Ger Apeldoorn <info@gerapeldoorn.nl>
 #
 class rhsm (
-  $rh_user             = undef,
-  $rh_password         = undef,
-  $org                 = undef,
-  $activationkey       = undef,
-  $pool                = undef,
-  $proxy_hostname      = undef,
-  $proxy_port          = undef,
-  $proxy_user          = undef,
-  $proxy_password      = undef,
-  $baseurl             = 'https://cdn.redhat.com',
-  $servername          = 'subscription.rhsm.redhat.com',
-  $serverprefix        = '/subscription',
-  $serverport          = 443,
-  $repo_ca_cert        = '%(ca_cert_dir)sredhat-uep.pem',
-  $manage_repos        = 1,
-  $full_refresh_on_yum = 0,
-  $package_ensure      = 'latest',
-  $repo_extras         = false,
-  $repo_optional       = false
+  $rh_user               = undef,
+  $rh_password           = undef,
+  $org                   = undef,
+  $activationkey         = undef,
+  $pool                  = undef,
+  $proxy_hostname        = undef,
+  $proxy_port            = undef,
+  $proxy_user            = undef,
+  $proxy_password        = undef,
+  $baseurl               = 'https://cdn.redhat.com',
+  $servername            = 'subscription.rhsm.redhat.com',
+  $serverprefix          = '/subscription',
+  $serverport            = 443,
+  $ca_cert_dir           = '/etc/rhsm/ca/',
+  $repo_ca_cert_filename = 'redhat-uep.pem',
+  $repo_ca_cert          = "%(ca_cert_dir)s${repo_ca_cert_filename}",
```

Comment:
How does this ```%(ca_cert_dir)s``` work? it looks like a python format string so is this a feature of rhsm?

Reply:
subscription-manager is Python based. The default setting is

```
repo_ca_cert = %(ca_cert_dir)sredhat-uep.pem
```

https://access.redhat.com/documentation/en-us/red_hat_subscription_management/1/html/rhsm/rhsm-config

Reply:
Ok. Using the variable here does work in puppet though I've seen kafo fail on it so I wonder what other tooling is out there that won't accept it. I'd be fine with accepting this but we may need to revisit it at some point.

Reply:
I thought about it as well. The other option would be not to use the "subscription-manager style" and to specify the full path, e.g.

```
$ca_cert_dir           = '/etc/rhsm/ca/',
$repo_ca_cert_filename = 'redhat-uep.pem',
```

```
# Server CA certificate location:
ca_cert_dir = <%= @ca_cert_dir %>

# Default CA cert to use when generating yum repo configs:
repo_ca_cert = <%= @ca_cert_dir %><%= @repo_ca_cert_filename %>
```

Reply:
Given you manage the file, I do think that would make sense. If they go out of sync, the code won't make sense anymore. You could even still use the ```%(repo_cert_dir)s``` in the template so that wouldn't change for users.